app/controllers/api_controller/reports.rb

```diff
@@ -38,5 +38,11 @@ def run_report_result(success, message = nil, options = {})
       add_task_to_result(res, options[:task_id]) if options[:task_id].present?
       res
     end
+
+    def import_resource_reports(_type, _id, data)
+      options = data.fetch("options", {}).symbolize_keys.merge(:userid => @auth_user)
+      result, meta = MiqReport.import_from_hash(data["report"], options)
+      action_result(true, meta[:message], :result => result)
```

Comment:
@abellotti done! let me know if/when you're ready to merge - I can squash some of these smaller commits